src/api/app/models/user.rb

```diff
@@ -768,20 +768,21 @@ def outgoing_requests(search = nil)
 
   # list of all requests
   def requests(search = nil)
-    bs_requests_table = BsRequest.arel_table
-    bs_actions_table = BsRequestAction.arel_table
-    reviews_table = Review.arel_table
-
-    result =
-      BsRequest.in_states(VALID_REQUEST_STATES).with_actions_and_reviews
-               .where(bs_requests_table[:creator].eq(login)
-                 .or(bs_actions_table[:target_project_id].in(involved_projects.pluck(:id)))
-                 .or(bs_actions_table[:target_package_id].in(involved_packages.pluck(:id)))
-                 .or(reviews_table[:state].eq(:new)
-                     .and(reviews_table[:user_id].eq(id)
-                     .or(reviews_table[:project_id].in(involved_projects.pluck(:id)))
-                     .or(reviews_table[:package_id].in(involved_packages.pluck(:id)))
-                     .or(reviews_table[:group_id].in(groups.pluck(:id))))))
+    projects = involved_projects.pluck(:id)
+    packages = involved_packages.pluck(:id)
+
+    # Test with production data if you change this query to make sure performance does not get worse
+    result = BsRequest.where(creator: login).or(
+      BsRequest.where(id: BsRequestAction.where(target_project_id: projects).select(:bs_request_id))).or(
+        BsRequest.where(id: BsRequestAction.where(target_package_id: packages).select(:bs_request_id))).or(
+          BsRequest.where(id: Review.where(state: :new, user_id: id).select(:bs_request_id)).or(
+            BsRequest.where(id: Review.where(state: :new, project_id: projects).select(:bs_request_id)).or(
+              BsRequest.where(id: Review.where(state: :new, package_id: packages).select(:bs_request_id)).or(
+                BsRequest.where(id: Review.where(state: :new, group_id: groups.pluck(:id)).select(:bs_request_id))
```

Comment:
Is there a way to make this code more readable? Maybe put these `.where` clauses into scopes?